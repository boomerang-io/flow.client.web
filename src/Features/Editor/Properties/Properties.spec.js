import React from "react";
import Inputs from ".";
import { screen, fireEvent } from "@testing-library/react";

const initialState = {};

const props = {
  summaryData: {
    properties: [
      {
        defaultValue: "pandas",
        description: "Tim parameter",
        key: "tim-parameter",
        label: "Tim parameter",
        required: true,
        type: "select",
        optiions: ["pandas", "dogs"],
      },
    ],
    id: "123",
  },
};

beforeEach(() => {
  document.body.setAttribute("id", "app");
});

describe("Inputs --- Snapshot Test", () => {
  it("Capturing Snapshot of Inputs", async () => {
    const { baseElement } = rtlContextRouterRender(<Inputs {...props} />, { initialState });
    expect(baseElement).toMatchSnapshot();
  });
});

describe("Inputs --- RTL", () => {
  it("Render inputs correctly", async () => {
    rtlContextRouterRender(<Inputs {...props} />, { initialState });
    expect(screen.getByText("tim-parameter")).toBeInTheDocument();
  });

  it("Opens create new parameter modal", async () => {
    rtlContextRouterRender(<Inputs {...props} />, { initialState });

    //expect(queryByText(/Create a new parameter/i)).not.toBeInTheDocument();

    //const modalTrigger = screen.getByText(/Create a new parameter/i);
    const modalTrigger = screen.getByTestId("create-parameter-button");
    fireEvent.click(modalTrigger);

    expect(screen.getByText(/Create a new parameter/i)).toBeInTheDocument();
  });

  it("Opens edit parameter modal", async () => {
    rtlContextRouterRender(<Inputs {...props} />, { initialState });

    //expect(queryByText(/Let's update it/i)).not.toBeInTheDocument();

    const modalTrigger = screen.getByLabelText(/Edit/i);
    fireEvent.click(modalTrigger);

    expect(screen.getByText(/Let's change some stuff/i)).toBeInTheDocument();
  });
});
