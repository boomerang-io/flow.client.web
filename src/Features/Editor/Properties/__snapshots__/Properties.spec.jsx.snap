// Vitest Snapshot v1

exports[`Inputs --- Snapshot Test > Capturing Snapshot of Inputs 1`] = `
<body
  id="app"
>
  <div>
    <div
      aria-label="Parameters"
      role="region"
    >
      <section>
        <div>
          <h1>
            Tim parameter
          </h1>
          <p>
            Tim parameter
          </p>
        </div>
        <dl>
          <dt>
            Key
          </dt>
          <dd>
            tim-parameter
          </dd>
        </dl>
        <dl>
          <dt>
            Type
          </dt>
          <dd>
            select
          </dd>
        </dl>
        <dl>
          <dt>
            Event Payload JsonPath
          </dt>
          <dd>
            ---
          </dd>
        </dl>
        <dl>
          <dt>
            Default value
          </dt>
          <dd>
            pandas
          </dd>
        </dl>
        <dl>
          <dt>
            Options
          </dt>
          <dd>
            ---
          </dd>
        </dl>
        <p>
          Required
        </p>
        <svg
          alt="Workflow edit button"
          aria-label="Edit"
          class=""
          data-testid="edit-parameter-button"
          role="button"
          style="will-change: auto;"
          tabindex="0"
          viewBox="0 0 26 26"
        >
          <title>
            edit
          </title>
          <g
            fill="none"
            fill-rule="evenodd"
            id="Symbols"
            stroke="none"
            stroke-width="1"
          >
            <g
              id="Editor-/-task-card"
              transform="translate(-190.000000, 13.000000)"
            >
              <g
                id="edit"
                transform="translate(191.000000, -12.000000)"
              >
                <circle
                  cx="12"
                  cy="12"
                  fill="#FFFFFF"
                  id="Oval"
                  r="12"
                  stroke="#0072C3"
                />
                <g
                  fill="#000000"
                  id="Actions-/-Formatting-/-edit-/-24"
                  transform="translate(4.000000, 4.000000)"
                >
                  <path
                    d="M1,13.5 L15,13.5 L15,14.5 L1,14.5 L1,13.5 Z M12.705,4.5 C12.892777,4.31243174 12.9982867,4.05790925 12.9982867,3.7925 C12.9982867,3.52709075 12.892777,3.27256826 12.705,3.085 L10.915,1.295 C10.7274317,1.10722296 10.4729092,1.00171334 10.2075,1.00171334 C9.94209075,1.00171334 9.68756826,1.10722296 9.5,1.295 L2,8.795 L2,12 L5.205,12 L12.705,4.5 Z M10.205,2 L12,3.795 L10.5,5.295 L8.705,3.5 L10.205,2 Z M3,11 L3,9.205 L8,4.205 L9.795,6 L4.795,11 L3,11 Z"
                    id="Fill"
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
        <svg
          alt="Workflow close button"
          class=""
          data-testid="workflow-delete-parameter-button"
          height="26px"
          role="button"
          tabindex="0"
          viewBox="0 0 26 26"
          width="26px"
        >
          <title>
            close-button
          </title>
          <g
            fill="none"
            fill-rule="evenodd"
            id="30-editor"
            stroke="none"
            stroke-width="1"
          >
            <g
              id="close-button"
              transform="translate(1.000000, 1.000000)"
            >
              <circle
                cx="12"
                cy="12"
                fill="#FFFFFF"
                id="Oval"
                r="12"
                stroke="#0072C3"
              />
              <g
                fill="#000000"
                id="Actions-/-Navigation-/-close-/-24"
                transform="translate(4.000000, 4.000000)"
              >
                <polygon
                  id="Fill"
                  points="12 4.7 11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8"
                />
              </g>
            </g>
          </g>
        </svg>
      </section>
      <button
        data-testid="create-parameter-button"
      >
        <div>
          <svg
            aria-label="Add"
            fill="currentColor"
            focusable="false"
            height="32"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
            />
          </svg>
          <p>
            Create a new parameter
          </p>
        </div>
      </button>
    </div>
  </div>
  <div
    class="bx--bmrg-modal-portal"
  />
  <div
    class="bx--bmrg-modal-portal"
  />
  <div
    class="bx--bmrg-modal-portal"
  />
</body>
`;
